sql/2020/03_Markup/pages_markup_by_device_and_button_types.sql

```diff
@@ -0,0 +1,44 @@
+#standardSQL
+# pages markup metrics grouped by device and button type
+
+# helper to create percent fields
+CREATE TEMP FUNCTION AS_PERCENT (freq FLOAT64, total FLOAT64) RETURNS FLOAT64 AS (
+  ROUND(SAFE_DIVIDE(freq, total), 4)
+);
+
+# returns all the data we need from _markup
+CREATE TEMPORARY FUNCTION get_markup_buttons_info(markup_string STRING)
+RETURNS ARRAY<STRUCT<
+name STRING,
+freq INT64
+>> LANGUAGE js AS '''
+var result = [];
+try {
+    var markup = JSON.parse(markup_string); 
+
+    if (Array.isArray(markup) || typeof markup != 'object') return result;
+
+    if (markup.buttons && markup.buttons.types) {
+      return Object.entries(markup.buttons.types).map(([name, freq]) => ({name, freq})); 
+    }
+
+} catch (e) {}
+return result;
+''';
+
+SELECT
+  _TABLE_SUFFIX AS client,
+  button_type_info.name AS button_type,
```

Comment:
If this value can be high cardinality, again I'd suggest LIMIT/HAVING.

Reply:
done